packages/sdks/src/types/components.ts

```diff
@@ -7,25 +7,70 @@ export interface ComponentInfo {
    * by registering a component with the same name, e.g. 'Text', to replace the built-in text component
    */
   name: string;
+  /** @hidden @deprecated */
   description?: string;
   /**
    * Link to a documentation page for this component
    */
   docsLink?: string;
+  /**
+   * Link to an image to be used as an icon for this component in Builder's editor
+   *
+   * @example
+   * ```js
+   * image: 'https://some-cdn.com/my-icon-for-this-component.png'
+   * ```
+   */
   image?: string;
+
+  // TODO (murtaza): Implement 'screenshot' related stuff
+  /**
+   * Link to a screenshot shown when user hovers over the component in Builder's editor
+   * use https://builder.io/upload to upload your screeshot, for easier resizing by Builder.
+   */
+  screenshot?: string;
+
+  // TODO (murtaza): Implement 'override' related stuff
+  /**
+   * When overriding built-in components, if you don't want any special behavior that
+   * the original has, set this to `true` to skip the default behavior
+   *
+   * Default behaviors include special "virtual options", such as a custom
+   * aspect ratio editor for Images, or a special column editor for Columns
+   *
+   * Learn more about overriding built-in components here: https://www.builder.io/c/docs/custom-components-overriding
+   */
+  override?: boolean;
+
   /**
    * Input schema for your component for users to fill in the options
    */
   inputs?: Input[];
+  /** @hidden @deprecated */
   class?: any;
+  /** @hidden @deprecated */
```

Comment:
Thanks for adding these tags. We will want to go ahead and completely remove any deprecated attributes in gen2 SDKs before we bump to v1, as this is our big chance to remove deprecated features.

They can stay here for now though, and these tags will help us know what to remove in the near future 👍🏽